shop/src/main/java/ro/msg/learning/shop/configuration/OrderStrategyConfiguration.java

```diff
@@ -0,0 +1,30 @@
+package ro.msg.learning.shop.configuration;
+
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import ro.msg.learning.shop.repository.StockRepository;
+import ro.msg.learning.shop.service.strategy.MostAbundantLocationStrategy;
+import ro.msg.learning.shop.service.strategy.OrderStrategy;
+import ro.msg.learning.shop.service.strategy.SingleLocationStrategy;
+
+
+@Configuration
+@RequiredArgsConstructor
+public class OrderStrategyConfiguration {
+    private static final String SINGLE_LOCATION = "singleLocation";
```

Comment:
Try to respect the Open-Closed principle here. Imagine that you would have to add a third option, how much effort would be included in your current solution? I would recommend creating an `enum` with the two strategy keys and using them within a switch case. This way if a third option would need to be added you would just add it in the enum and a new case within the switch. Of course besides the implementation itself

Reply:
cool trick. loved refactoring it.